src/internal_macros.rs

```diff
@@ -564,3 +564,22 @@ macro_rules! user_enum {
         }
     );
 }
+
+/// Formats error. If `std` feature is OFF appends error source (delimited by `: `). We do this
+/// because `e.source()` is only available in std builds, without this macro the error source is
+/// lost for no-std builds.
+macro_rules! write_err {
+    ($writer:expr, $string:literal $(, $args:expr),*; $source:expr) => {
+        {
+            #[cfg(feature = "std")]
+            {
+                let _ = &$source;   // Prevents clippy warnings.
+                write!($writer, "{}", format_args!($string $(, $args)*))
```

Comment:
I believe, in `ln-types` I used `format_args!` to implement some trick around no-alloc, which we don't need here, so we should be able to just `write!($writer, $string $(, $args)*)`.

Similarly below `write!($writer, concat!($string, ": {}") $(, $args)*, $source)`

Reply:
Ahh, cheers. Used as suggested.